### PR TITLE
DT 495 Framework for NHS Sync (WIP)

### PR DESCRIPTION
**Pull request by Mjwillis**
Will be adding tests and more functionality next week!

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  implementation("io.springfox:springfox-swagger2:2.9.2")
+  implementation("io.springfox:springfox-swagger-ui:2.9.2")
+  implementation("io.springfox:springfox-bean-validators:2.9.2")
+
+  implementation( "com.google.code.gson:gson:2.8.6")
+
+  implementation("org.springframework:spring-jms")
+  implementation("com.amazonaws:amazon-sqs-java-messaging-lib:1.0.8")
+//  implementation platform("com.amazonaws:aws-java-sdk-bom:1.11.735")
```
Comment (andrewrlee) [inline build.gradle.kts]: guess this can go

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt [hunk 2]
```diff
@@ -1,80 +1,80 @@
...
-    val defaultTokenServices = DefaultTokenServices()
-    defaultTokenServices.setTokenStore(tokenStore())
-    return defaultTokenServices
-  }
-
-  @Bean
-  @ConfigurationProperties("prisontonhs.client")
-  open fun prisonToNhsClientCredentials(): ClientCredentialsResourceDetails = ClientCredentialsResourceDetails()
+    @Bean
+    open fun api(): Docket? {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt]: I think a lot of place like this returns nullable unnecessary?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerPatientUpdateService.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(this::class.java)
+        val gson: Gson = GsonBuilder().setPrettyPrinting()
+                .registerTypeAdapter(LocalDate::class.java, LocalDateAdapter())
+                .create();
+    }
+
+    fun externalMovement(externalMovement: ExternalPrisonerMovementMessage) {
+
+        if (externalMovement.movementType in listOf("ADM", "REL")) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerPatientUpdateService.kt]: I suspect others may know what **ADM** and **REL** mean.  Sorry I don't.  I would find it helpful is these were pulled out into constants...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/NhsReceiveService.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import java.time.Duration
+
+@Service
+open class NhsReceiveService(@Qualifier("webClient") val webClient: WebClient,
+                             @Value("\${api.base.url.nhs}") val baseUri: String) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  private val timeout: Duration = Duration.ofSeconds(30)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/NhsReceiveService.kt]: This is done in a few places.  Make configurable?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerChangeListenerPusher.kt [hunk 5]
```diff
@@ -9,25 +9,26 @@ import org.springframework.stereotype.Service
...
+    private val prisonerPatientUpdateService: PrisonerPatientUpdateService
 ) {
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
     val gson: Gson = GsonBuilder().create()
   }
 
   @JmsListener(destination = "\${sqs.queue.name}")
-  open fun pushPrisonUpdateToProbation(requestJson: String?) {
+  open fun pushPrisonUpdateToxProbation(requestJson: String?) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerChangeListenerPusher.kt]: Is the 'x' in the method name intentional?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt (hunk 2 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt]: changed

File: build.gradle.kts (hunk 1 above)
Comment (Mjwillis) [reply build.gradle.kts]: updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerPatientUpdateService.kt (hunk 3 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerPatientUpdateService.kt]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/NhsReceiveService.kt (hunk 4 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/NhsReceiveService.kt]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerChangeListenerPusher.kt (hunk 5 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/PrisonerChangeListenerPusher.kt]: typo!

**Pull request merged**